packages/frontend/src/components/json_import.tsx

```diff
@@ -0,0 +1,106 @@
+import { createSignal } from "solid-js";
+import type { Document } from "../api";
+import "./json_import.css";
+
+interface JsonImportProps<T extends string> {
+    onImport: (data: Document<T>) => void;
+    validate?: (data: Document<T>) => boolean | string;
+}
+
+export const JsonImport = <T extends string>({ onImport, validate }: JsonImportProps<T>) => {
```

Comment:
I think having a doc-comment on the top-level component/primary export is a nice touch. It doesn't have to be long.